schemas/org.gnome.shell.extensions.dash-to-panel.gschema.xml

```diff
@@ -372,6 +372,42 @@
       <summary>Animate Show Applications from the desktop</summary>
       <description>Animate Show Applications from the desktop</description>
     </key>
+    <key type="as" name="show-apps-button-context-menu-commands">
+     <default>['gnome-control-center power',
+         'gnome-logs',
+         'gnome-control-center info-overview',
+         'gnome-control-center display',
+         'gnome-disks']</default>
+     <summary>Commands to run on Show Apps Menu</summary>
+     <description>Commands to run in the Show Apps button context menu</description>
+    </key>
+    <key type="as" name="show-apps-button-context-menu-titles">
+     <default>['Power options',
+         'Event logs',
```

Comment:
Sorry, I should have though of that before, but we didn't setup make/gettext to extract translatable strings from the gschema.xml file (at the moment all the localized texts are in the .js and Settings.ui files). [It is possible](https://developer.gnome.org/gio/stable/ch34s06.html) to add that, but I think the best course of action would be to somehow have those labels in the Settings.ui file, as this is where they'll ultimately end up (when we add the ui to allow users to edit their context menu entries). Maybe in a hidden listbox or something... I'll check if I can find a control that would lends itself for this.

Reply:
Oh man, I see what you mean. By putting the strings in the schema, gsettings won't be able to pull the localized texts. Here's some options:
* Option 1: Keep the new menu arrays empty and bring back the hardcoded entries and don't allow customization of their commands. If the default command doesn't exist (#847) on the system, "checkIfCommandExists" will remove it quietly. Users can then use the new arrays to add new menu entries to replace the missing entries, or add their own as they like.
* Option 2: I could add a new setting for each of the default localized entries like this:
**show-apps-button-context-power-options**: gnome-control-center power
**show-apps-button-context-event-logs**: gnome-logs
**show-apps-button-context-system**: gnome-control-center info-overview
...
and then I could keep the **show-apps-button-context-menu-commands** values for custom strings. This is the same as option 1, but instead of hard coding the command, I check for the specific setting value. A user could then change the default entries and add additional custom entries with dconf or gsettings...

FWIW, here's the dconf commands I'm using for testing this.

```
dconf write /org/gnome/shell/extensions/dash-to-panel/show-apps-button-context-power-options '["gnome-control-center power-version2"]'
dconf write /org/gnome/shell/extensions/dash-to-panel/panel-context-menu-commands '["gnome-terminal"]'
dconf write /org/gnome/shell/extensions/dash-to-panel/panel-context-menu-commands '["gnome-terminal"]'
```

With either option, we can keep the localized text in the js and move them to the UI when it makes sense. I like the first option because it introduces fewer settings to support in the future. What do you think?

Reply:
Good idea, option 1 sounds good to me! Should be perfect if the gsettings array can be used to override the hardcoded entries. With these changes, I'm not even sure that adding a UI to edit these entries will be necessary as the minority of users who will need such an option will be able to use dconf. Great!

Reply:
Once this is in, it would be nice to add some sample commands to the wiki to document it. Here's some suggestions.

To add a command when clicking on the apps button:

* `dconf write /org/gnome/shell/extensions/dash-to-panel/panel-context-menu-commands '["gnome-terminal"]'`
* `dconf write /org/gnome/shell/extensions/dash-to-panel/panel-context-menu-titles '["terminal app"]'`

To add a command on the panel:

* `dconf write /org/gnome/shell/extensions/dash-to-panel/show-apps-button-context-menu-commands '["gedit", "gnome-calculator"]'`
* `dconf write /org/gnome/shell/extensions/dash-to-panel/show-apps-button-context-menu-titles '["Text Editor", "Calculator"]'`